src/datreant/metadata.py

```diff
@@ -87,12 +87,16 @@ def _getselection(self, value):
             elif isinstance(value, tuple):
                 # a tuple of tags gives members with ANY of the tags
                 fits = any(self._getselection(item) for item in value)
-            if isinstance(value, set):
+            elif isinstance(value, set):
                 # a set of tags gives only members WITHOUT ALL the tags
                 # can be used for `not`, basically
                 fits = not all(self._getselection(item) for item in value)
             elif isinstance(value, str):
                 fits = value in self
+            else:
+                raise ValueError(
+                        "Given value is not a valid selection. "
```

Comment:
It might be useful to explicitly show in the error message what `value` was (especially its `type`), maybe
```python
"The given value {value} of type {type_value} is not a valid selection. ...".format(value=repr(value), type_value=type(value))
```